src/python_routing_v02/fast_reach/mc_reach.pyx

```diff
@@ -550,49 +603,49 @@ cpdef object compute_network_multithread(int nsteps, list reaches, dict connecti
                         buf_view[drows[i], 10] = initial_conditions[srows[i],1]
                         buf_view[drows[i], 11] = 0.0
                         buf_view[drows[i], 12] = initial_conditions[srows[i],2]
-                
+
                 # ------ !!!!! MULTITHREAD LOOP !!!!! ------ #
                 # compute each reach
                 for r in prange(istart,iend+1):
-                    
+
                     # reach length for reach r of group
                     ireach_cache = ireach_cache_array[r]
                     reachlen = -reach_cache[ireach_cache]
-                    
+
                     # total reach length of reaches preceeding reach r in group
                     if r > istart:
                         prevreachlen = 0
                         for i in range((r-istart)):
                             prevreachlen = prevreachlen + -reach_cache[ireach_cache_array[(r-(i+1))]]
                     else:
                         prevreachlen = 0
-                            
+
                     # compute reach routing
-                    if assume_short_ts:    
-                        compute_reach_kernel(qup_view[r-istart], 
+                    if assume_short_ts:
```

Comment:
Why are we branching here?

Reply:
@awlostowski-noaa 
Should this be: 
```
                    if assume_short_ts:
                        compute_reach_kernel(qup_view[r-istart],
                                             qup_view[r-istart], # quc = qup
                                             reachlen,
                                             buf_view[prevreachlen:prevreachlen+reachlen,:],
                                             out_view[prevreachlen:prevreachlen+reachlen,:],
                                             assume_short_ts)
```

Reply:
@groutr 
This compute_network_MULTITHREAD implementation was an experiment in containing the looping entirely within cython -- we have stepped back from that effort for the moment and the latest development effort has been using the original compute_network function call, with minor modifications to accept upstream inflows.